src/report.rs

```diff
@@ -0,0 +1,356 @@
+use std::path::PathBuf;
+
+use serde::Serialize;
+
+use crate::{abis::*, full_test_name, WriteBuffer};
+
+/// These are the builtin test-expectations, edit these if there are new rules!
+pub fn get_test_rules(test: &TestKey, caller: &dyn AbiImpl, callee: &dyn AbiImpl) -> TestRules {
+    use TestCheckMode::*;
+    use TestRunMode::*;
+
+    // By default, require tests to run completely and pass
+    let mut result = TestRules {
+        run: Check,
+        check: Pass(Check),
+    };
+
+    // Now apply specific custom expectations for platforms/suites
+    let is_c = caller.lang() == "c" || callee.lang() == "c";
+    let is_rust = caller.lang() == "rust" || callee.lang() == "rust";
+    let is_rust_and_c = is_c && is_rust;
+
+    // llvm and gcc disagree on the u128 ABI everywhere but aarch64 (arm64).
+    // This is Bad! Ideally we should check for all clang<->gcc pairs but to start
+    // let's mark rust <-> C as disagreeing (because rust also disagrees with clang).
+    if !cfg!(target_arch = "aarch64") && test.test_name == "ui128" && is_rust_and_c {
+        result.check = Busted(Check);
+    }
+
+    // THIS AREA RESERVED FOR VENDORS TO APPLY PATCHES
```

Comment:
Maybe you could make the start and end comment a couple of lines longer so patches won't include lines outside of this section as part of the context? That should make them a bit more robust.